Sources/tart/VM.swift

```diff
@@ -392,12 +392,19 @@ class VM: NSObject, VZVirtualMachineDelegate, ObservableObject {
 
     // Version console device
     //
-    // A dummy console device useful for implementing
+    // A console device useful for implementing
     // host feature checks in the guest agent software.
     if !suspendable {
+      let consoleURL: URL = URL(fileURLWithPath: "tart-agent.sock", isDirectory: false, relativeTo: diskURL).absoluteURL
       let consolePort = VZVirtioConsolePortConfiguration()
-      consolePort.name = "tart-version-\(CI.version)"
 
+      if FileManager.default.fileExists(atPath: consoleURL.absoluteURL.path()) == false {
+        FileManager.default.createFile(atPath: consoleURL.absoluteURL.path(), contents: nil)
```

Comment:
It seems to me that this won't create a socket, just a regular file 🤔

Reply:
It's a virtio socket not an unix domain socket.

It's not necessary to set some control flags like unix socket and if it will be necessary it will specified in the apple doc....

Reply:
see below an example.

After it's possible to implement some protocoled agent like gRPC, http/https.....
it's a pipe.

Reply:
>it's a pipe.

I'm sorry, but it's not a pipe. Nor a socket, which you call it in https://github.com/cirruslabs/tart/pull/978#issue-2748210766 and by giving it a `.sock` extension.

It's a regular file stored on disk.

Moreover, you're using it for both for reading and writing, which will mess up the file contents badly if you'd ever want to implement any bidirectional communication on top of it.

Reply:
> Moreover, you're using it for both for reading and writing, which will mess up the file contents badly if you'd ever want to implement any bidirectional communication on top of it.

I don't see really the problem because it's a feature enabled by VZ. With a unix domain socket too you have the same problem.

You can write garbage to socket also, the effect is the same.

So it's could be replace by a tty...